textual_fspicker/parts/directory_navigation.py

```diff
@@ -305,8 +306,7 @@ def _settle_highlight(self) -> None:
     @property
     def is_root(self) -> bool:
         """Are we at the root of the filesystem?"""
-        # TODO: Worry about portability.
-        return self._location == MakePath.of(self._location.root)
+        return self._location == MakePath.of(self._location.parent)
```

Comment:
Fun! I think I just realised what the actual mistake might have been here. Consider:

```python
>>> type(Path("/foo").root)
<class 'str'>
>>> type(Path("/foo").parent)
<class 'pathlib.PosixPath'>
>>>
```

So I'm curious now, on Windows, would this also work:

```python
return Path(self._location) == MakePath.of(self._location.root)
```

Reply:
Unfortunately, no, because `.root` does not include the drive:
```python
>>> pathlib.Path("c:\\").root
'\\'
```

Reply:
All things considered, I thought that the 'root is its own parent' trick is easier to understand then trying to compare to the root with the drive included, but only on Windows.

Reply:
Agreed, I much prefer your approach too. I was just curious as to how it played out on Windows. :-)